### PR TITLE
Update cudf dependency to 0.14

### PR DESCRIPTION
**Pull request by jclay**
Also updates ci script to use the environment.yml file.

### CODE REVIEW

**Comment by jclay** [comment]
@andrewseidl would appreciate a review on this when you have a chance. I believe we decided this will become `v0.24.1`. Let me know if you see any issues with the update to the ci script.

File: ci/install-test-deps-conda.sh [hunk 1]
```diff
@@ -4,36 +4,12 @@ echo
...
-pytest-cov \
-pytest-mock \
-shapely \
-sphinx  \
-requests \
-sphinx_rtd_theme \
-'rbc==0.2.2'
+# create a copy of the environment file, replacing
+# with the python version we specify.
+sed -E "s/python.+$/python=$PYTHON/" ./environment_gpu.yml > /tmp/environment_gpu_${PYTHON}.yml
```
Comment (andrewseidl) [inline ci/install-test-deps-conda.sh:10]: Should this be a little more explicit? For example `- python[^:alpha:]+$`

File: ci/install-test-deps-conda.sh [hunk 2]
```diff
@@ -4,40 +4,16 @@ echo
...
-pytest-cov \
-pytest-mock \
-shapely \
-sphinx  \
-requests \
-sphinx_rtd_theme \
-'rbc==0.2.2'
+# create a copy of the environment file, replacing
+# with the python version we specify.
+sed -E "s/- python[^[:alpha:]]+$/- python=$PYTHON/" ./environment_gpu.yml > /tmp/environment_gpu_${PYTHON}.yml
```
Comment (mikehinchey) [inline ci/install-test-deps-conda.sh:10]: Is it good to use /tmp? Is there a ./target dir that would work for a tmp file?

Comment (jclay) [reply ci/install-test-deps-conda.sh:10]: This is just for the CI suite so I don't see any issues. My guess is it gets cleared anyways when the container resets between runs. @andrewseidl does that sound correct? 

Sadly we're not able to pipe into conda with, for example, `cat env.yml | conda env create  -f  -` as that would solve the problem more eloquently.

Comment (andrewseidl) [reply ci/install-test-deps-conda.sh:10]: Yup, it's run inside a container, so it gets cleared for every build. If we want to be really sure we could use `mktemp` to give us a temp dir, but that's probably not necessary.

**Pull request merged**